source/common/common/key_value_store_base.cc

```diff
@@ -51,14 +61,32 @@ bool KeyValueStoreBase::parseContents(absl::string_view contents) {
   while (!contents.empty()) {
     absl::optional<absl::string_view> key = getToken(contents, error);
     absl::optional<absl::string_view> value;
+    absl::optional<std::chrono::seconds> ttl = absl::nullopt;
     if (key.has_value()) {
       value = getToken(contents, error);
     }
     if (!key.has_value() || !value.has_value()) {
       ENVOY_LOG(warn, error);
       return false;
     }
-    addOrUpdate(key.value(), value.value(), absl::nullopt);
+    if (checkForTtl(contents)) {
+      uint64_t ttl_int;
+      auto token = getToken(contents, error);
+      if (token.has_value() && absl::SimpleAtoi(token.value(), &ttl_int)) {
+        ttl.emplace(std::chrono::duration_cast<std::chrono::seconds>(
+            std::chrono::time_point<std::chrono::steady_clock>(std::chrono::seconds(ttl_int)) -
+            time_source_.monotonicTime()));
+      } else {
+        if (!token.has_value()) {
+          ENVOY_LOG(warn, error);
+        } else {
+          ENVOY_LOG(warn, "TTL was read from disk but failed to convert to integer");
+        }
+      }
+    }
+    if (!ttl.has_value() || ttl > std::chrono::seconds(0)) {
```

Comment:
Would we ever encode a 0 TTL?

Reply:
These TTLs are being calculated from disk.  So in theory if we happen to load a key from disk the exact second it's TTL expires, we should drop it.